packages/hoprd/src/logs.ts

```diff
@@ -1,12 +1,44 @@
 import ws from 'ws'
 import debug from 'debug'
+import RunQueue from 'run-queue'
+import { randomBytes } from 'crypto'
+import { Ed25519Provider } from 'key-did-provider-ed25519'
+import KeyResolver from 'key-did-resolver'
+import { DID } from 'dids'
+import CeramicClient from '@ceramicnetwork/http-client'
+import { TileDocument } from '@ceramicnetwork/stream-tile'
 
 export type Socket = ws
 
 let debugLog = debug('hoprd')
 
 const MAX_MESSAGES_CACHED = 100
 
+class Queue {
+  private queue: RunQueue = undefined
+  private completed: boolean = false
+  private opts: {
+    maxConcurrency: number
+  } = undefined
+
+  constructor(opts) {
+    this.opts = opts
+    this.queue = new RunQueue(opts)
+  }
+  add(priority, job) {
+    if (this.completed) {
+      this.queue = new RunQueue(this.opts)
+      this.completed = false
+    }
+    this.queue.add(priority, job)
+  }
+  async run() {
+    this.queue.run().then(() => (this.completed = true))
+  }
+}
+
+const queue = new Queue({ maxConcurrency: 1 })
```

Comment:
This should be configurable.

Reply:
Makes no sense for the current context of the `queue`, we want it to be always `1` to ensure the commits to the sink are made in order.